eos-reclaim-swap

```diff
@@ -0,0 +1,122 @@
+        cmd = ['fdisk', '-l', disk]
+        out = subprocess.run(cmd, stdout=subprocess.PIPE).stdout
+        return out.decode()
+
+
+    def find_disk(self, partition):
+        '''
+        Return path to a block device representing the disk containing
+        partition.
+        '''
+        p = os.path.realpath(partition)
+        disk = None
+
+        # /dev/sd[a-z][0-9]
+        if (p[:-2] == '/dev/sd'):
+            disk = p[:-1]
+            self.sep = ''
+
+        # /dev/[a-z]+p[0-9], but not /dev/sdp[0-9] (eMMC)
+        elif (p.startswith('/dev/') and p[-2:-1] == 'p'):
+            disk = p[:-2]
+            self.sep = 'p'
+
+        # device-mapped images (dual-boot setups)
+        if (p[-1] == '/dev/mapper/endless-image'):
+            disk = partition[:-1]
+            self.sep = ''
+
+        # for testing with loop devices
+        if (p[:-3] == '/dev/loop'):
+            self.sep = 'p'
+
+        # for testing with image files
+        if not p.startswith('/dev/'):
+            disk = partition[:-1]
+            self.sep = ''
+
+        return disk
+
+
+    def reclaim_swap_space(self, disk_path):
+        '''
+        Remove the first swap partition on disk and expand the preceding
+        partition so it spawns over the space previously used by the swap
+        partition.
+        '''
+        dev = parted.getDevice(disk_path)
+        disk = parted.newDisk(dev)
+        parts = disk.partitions
+
+        for p in range(len(parts)):
+            # (py)parted does not expose the MBR partition ID or the GPT GUID,
+            # so we need to use the file system signature to identify swap
+            # partitions and thus this only works with formatted swap
+            # partitions. When the fileSystem property is set on swap
+            # partitions it can be either linux-swap(v0) or linux-swap(v1).
+            if parts[p].fileSystem and parts[p].fileSystem.type[:-4] == 'linux-swap':
+                geom = parted.Geometry(start=parts[p-1].geometry.start,
+                                       end=parts[p].geometry.end, device=dev)
+                constraint = parted.Constraint(maxGeom=geom)
+                disk.removePartition(parts[p])
+                disk.maximizePartition(parts[p-1], constraint)
+                disk.commit()
+
+                return True
+
+        return False
+
+
+    def find_disk_and_reclaim_swap_space(self, part, unit):
+        if not part:
+            part = subprocess.run(['systemctl', 'show', '-p', 'What', unit],
+                                  stdout=subprocess.PIPE).stdout
+            part = part.decode()
+            part = part.split(sep='=')[1].strip()
+
+        disk = self.find_disk(part)
+        if (disk == None):
+            raise ValueError('Could not find disk for partition %s' % part)
+        if self.verbose: print('Old disk table:\n' + self.get_table_str(disk))
+
+        removed = self.reclaim_swap_space(disk)
+        if (removed and os.geteuid() == 0):
```

Comment:
I don't think making this decision based on `os.geteuid` is a great idea: I habitually run this test suite as root (in a VM of course), so every time I run it this block will run, and the second and subsequent times it runs, this block will spuriously fail because `/var/eos-booted` does not exist. I suggest some of the following:

1. change this condition to be based on whether `--partition` was passed, which will prevent this block running from the test suite (for better or for worse)
2. catch and ignore `FileNotFoundException` from `os.remove`

Reply:
Catching the exception here actually makes sense for production as well, as if `/var/eos-booted` is not present the filesystem resize is effectively already scheduled for the next boot. I think we should still only try to remove the file if running as root, and suppressing `FileNotFoundError` should be enough to cover your use-case.